gr-blocks/lib/plateau_detector_fb_impl.cc

```diff
@@ -86,6 +86,16 @@ namespace gr {
       return i;
     }
 
+    void plateau_detector_fb_impl::set_threshold(float threshold)
+    {
+      d_threshold = threshold;
```

Comment:
We should think about thread safety here. What happens if we change the threshold midst-`general_work`?

For now, this is as fine as "any other block<sup>TM</sup>", but if I'm brutally honest, "any other block<sup>TM</sup>" is still broken.

Anyway, not anything that I plan to fix within the scope of this PR, so this is fine to me :)

Reply:
Indeed thread safety wasn't paramount. Adding in the mutex from gr::block is pretty trivial if you want me to do that.

Reply:
But, yes, many (most?) GR blocks aren't truly thread-safe when setting variables. "there has got to be a better way"™ ...

Reply:
yep, exactly, I have Ideas<sup>TM</sup> about what that better way<sup>TM</sup> should look like, but we can't proactively fix before we've discussed that approach.

If you feel it's really not much work, do add the mutex here. It's the Right Thing<sup>TM</sup> to do :)